pmd-apex/src/main/resources/category/apex/bestpractices.xml

```diff
@@ -64,16 +64,27 @@ public class Foo {
     </rule>
 
     <rule name="ApexUnitTestMethodShouldHaveIsTestAnnotation"
-          language="apex"
           since="6.13.0"
+          language="apex"
           message="Apex test methods should have @isTest annotation."
-          class="net.sourceforge.pmd.lang.apex.rule.bestpractices.ApexUnitTestMethodShouldHaveIsTestAnnotationRule"
+          class="net.sourceforge.pmd.lang.apex.rule.ApexXPathRule"
           externalInfoUrl="${pmd.website.baseurl}/pmd_rules_apex_bestpractices.html#apexunittestmethodshouldhaveistestannotation">
         <description>
-Apex test methods should have @isTest annotation.
-As testMethod keyword is deprecated, Salesforce advices to use @isTest annotation for test class/methods.
+Apex test methods should have `@isTest` annotation instead of the `testMethod` keyword,
+as `testMethod` is deprecated.
+Salesforce advices to use [@isTest](https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/apex_classes_annotation_isTest.htm)
+annotation for test classes and methods.
         </description>
         <priority>3</priority>
+        <properties>
+            <property name="xpath">
```

Comment:
You probably also need the following property here:

```xml
<property name="version" value="2.0"/>
```

That might be the reason, why you see different results when running in the designer and running the tests. The designer might use by default XPath 2.0.

Reply:
nailed it!